tdrs-frontend/src/components/Reports/ReportsContext.jsx

```diff
@@ -20,35 +28,151 @@ export const useReportsContext = () => {
   return context
 }
 
+// Valid file types for each report mode
+const VALID_FILE_TYPES = {
+  reports: [
+    'tanf',
+    'ssp-moe',
+    ...(process.env.REACT_APP_SHOW_PIA === 'true'
+      ? ['program-integrity-audit']
+      : []),
+  ],
+  fra: [
+    'workOutcomesOfTanfExiters',
+    // TODO: uncomment when we collect this data
+    // 'secondarySchoolAttainment',
+    // 'supplementalWorkOutcomes',
+  ],
+}
+
+// Valid quarters
+const VALID_QUARTERS = Object.keys(quarters) // ['Q1', 'Q2', 'Q3', 'Q4']
+
+/**
+ * Validates URL parameters for the report pages.
+ * Invalid parameters are cleared individually, valid ones are kept.
+ * @param {URLSearchParams} searchParams - URL search parameters
+ * @param {boolean} isFra - Whether this is for FRA reports
+ * @param {Array} sttList - List of valid STTs
+ */
+const validateUrlParams = (searchParams, isFra, sttList) => {
+  const fiscalYear = searchParams.get('fy')
+  const quarter = searchParams.get('q')
+  const type = searchParams.get('type')
+  const stt = searchParams.get('stt')
+  const tab = searchParams.get('tab')
+
+  const defaultType = isFra ? 'workOutcomesOfTanfExiters' : 'tanf'
+  let validatedFy = ''
+  let validatedQ = ''
+  let validatedType = defaultType
+  let validatedStt = ''
+  let validatedTab = 1
+  let hasInvalidParam = false
+
+  // Validate fiscal year (must be a valid year between 2021 and current fiscal year)
+  if (fiscalYear) {
+    const year = parseInt(fiscalYear, 10)
+    const currentFiscalYear = getCurrentFiscalYear()
+    const minYear = 2021
```

Comment:
i was able to get a weird state with pia and playing with the year parameter
<img width="1104" height="960" alt="image" src="https://github.com/user-attachments/assets/8b78e18e-cf48-4019-8b6d-4191db841ec5" />